jwst/associations/lib/rules_level3_base.py

```diff
@@ -294,6 +299,50 @@ def make_member(self, item):
         }
         return member
 
+    def make_fixedslit_bkg(self):
+        """Add a background to a MIR_lrs-fixedslit observation"""
+        for product in self['products']:
+            members = product['members']
+
+            # Split out the science vs. non-science
+            # The non-science exposures will get attached
+            # to every resulting association.
+            science_exps = [
+                member
+                for member in members
+                if member['exptype'] == 'science'
+            ]
+            nonscience_exps = [
```

Comment:
Since the code is now only appending to the current member list, this section, I believe, is unnecessary. If anything, it may be doubling up the non-science members, see below.

Reply:
removed